packages/salesforcedx-vscode-core/src/commands/forceApexClassCreate.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import glob = require('glob');
+import { nls } from '../messages';
+import {
+  CancelResponse,
+  CompositeParametersGatherer,
+  ContinueResponse,
+  DirFileNameSelection,
+  ParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+class SelectFileName implements ParametersGatherer<{ fileName: string }> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string }>
+  > {
+    const fileNameInputOptions = <vscode.InputBoxOptions>{
+      prompt: nls.localize('force_apex_class_create_enter_file_name')
+    };
+    const fileName = await vscode.window.showInputBox(fileNameInputOptions);
+    return fileName
+      ? { type: 'CONTINUE', data: { fileName } }
+      : { type: 'CANCEL' };
+  }
+}
+
+class SelectDirPath implements ParametersGatherer<{ outputdir: string }> {
+  private explorerDir: string | undefined;
+
+  public constructor(explorerDir?: { path: string }) {
+    this.explorerDir = explorerDir ? explorerDir.path : explorerDir;
+  }
+
+  public globDirs(srcPath: string, prioritize?: string): string[] {
+    const unprioritizedDirs = new glob.GlobSync(srcPath + '/**/').found;
+    if (prioritize) {
+      const notPrioritized: string[] = [];
+      const prioritized = unprioritizedDirs.filter(dir => {
+        if (dir.includes(prioritize)) {
+          return true;
+        } else {
+          notPrioritized.push(dir);
+        }
+      });
+      return prioritized.concat(notPrioritized);
+    }
+    return unprioritizedDirs;
+  }
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ outputdir: string }>
+  > {
+    const rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : '';
```

Comment:
Is it reasonable to replace that with `const rootPath = vscode.workspace.rootPath!;` ? The extension/commands are only active when there is a workspace and it contains sfdx-project.json